android-agent/src/main/kotlin/io/opentelemetry/android/agent/AndroidAgent.kt

```diff
@@ -0,0 +1,167 @@
+ */
+
+package io.opentelemetry.android.agent
+
+import android.app.Application
+import io.opentelemetry.android.OpenTelemetryRum
+import io.opentelemetry.android.OpenTelemetryRumBuilder
+import io.opentelemetry.android.agent.endpoint.EndpointConfig
+import io.opentelemetry.android.agent.session.SessionIdTimeoutHandler
+import io.opentelemetry.android.agent.session.SessionManager
+import io.opentelemetry.android.config.OtelRumConfig
+import io.opentelemetry.android.features.diskbuffering.DiskBufferingConfiguration
+import io.opentelemetry.android.instrumentation.AndroidInstrumentationLoader
+import io.opentelemetry.android.instrumentation.activity.ActivityLifecycleInstrumentation
+import io.opentelemetry.android.instrumentation.anr.AnrInstrumentation
+import io.opentelemetry.android.instrumentation.common.ScreenNameExtractor
+import io.opentelemetry.android.instrumentation.crash.CrashDetails
+import io.opentelemetry.android.instrumentation.crash.CrashReporterInstrumentation
+import io.opentelemetry.android.instrumentation.fragment.FragmentLifecycleInstrumentation
+import io.opentelemetry.android.instrumentation.network.NetworkChangeInstrumentation
+import io.opentelemetry.android.instrumentation.slowrendering.SlowRenderingInstrumentation
+import io.opentelemetry.android.internal.services.ServiceManager
+import io.opentelemetry.android.internal.services.network.data.CurrentNetwork
+import io.opentelemetry.api.trace.Tracer
+import io.opentelemetry.exporter.otlp.http.logs.OtlpHttpLogRecordExporter
+import io.opentelemetry.exporter.otlp.http.trace.OtlpHttpSpanExporter
+import io.opentelemetry.instrumentation.api.instrumenter.AttributesExtractor
+import io.opentelemetry.sdk.common.Clock
+import java.time.Duration
+
+object AndroidAgent {
+    private val activityLifecycleInstrumentation by lazy {
+        AndroidInstrumentationLoader.getInstrumentation(
+            ActivityLifecycleInstrumentation::class.java,
+        )!!
+    }
+    private val fragmentLifecycleInstrumentation by lazy {
+        AndroidInstrumentationLoader.getInstrumentation(
+            FragmentLifecycleInstrumentation::class.java,
+        )!!
+    }
+    private val anrInstrumentation by lazy {
+        AndroidInstrumentationLoader.getInstrumentation(
+            AnrInstrumentation::class.java,
+        )!!
+    }
+    private val crashReporterInstrumentation by lazy {
+        AndroidInstrumentationLoader.getInstrumentation(
+            CrashReporterInstrumentation::class.java,
+        )!!
+    }
+    private val networkChangeInstrumentation by lazy {
+        AndroidInstrumentationLoader.getInstrumentation(NetworkChangeInstrumentation::class.java)!!
+    }
+    private val slowRenderingInstrumentation by lazy {
+        AndroidInstrumentationLoader.getInstrumentation(SlowRenderingInstrumentation::class.java)!!
+    }
+
+    fun createRumBuilder(
+        application: Application,
+        otelRumConfig: OtelRumConfig = OtelRumConfig(),
```

Comment:
as a user, when I see that I can pass in a "rum config" object, I instantly wonder what's in there. ...but then I see this other stuff alongside it and wonder even more why some of this configurable stuff is in the config but some of it is passed directly to the builder. Seems inconsistent.

Reply:
That's true, although I'd argue that we're already in a similar situation by having an `OtelRumConfig` object passed to the `OpenTelemetryRumBuilder` which itself has some config options too! 😅 

At some point, I was thinking about `OtelRumConfig` as a place for flags so that we could switch things on/off and also provide a serializable part of the config which might be useful for debug logs, and then the rest of config options in the builder as a place for more complex objects... Although maybe we would also like to show something else in the debug logs apart from just flags, in which case I'm not sure how can I justify having both tbh. What's your view on this?